src/librustc_typeck/check/method/suggest.rs

```diff
@@ -173,7 +173,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                      if mode == Mode::MethodCall {
                                          "method"
                                      } else {
-                                         "associated item"
+                                         match item_name.as_str().chars().next() {
+                                             Some(name) => {
+                                                 if name.is_lowercase() {
+                                                     "function or associated item"
```

Comment:
I it were a const, you'd need to wrap it in parens to be able to do a function call, and the const must be a Fn* or fn type